commit c1ae07116d9bebbe580b3df22f8c8f3a7cbeec44
Author: Kevin Williams <admin@utermux.dev>
Date:   Sat Dec 23 16:47:56 2023 +0800

    Revert "cmake: only install public headers"
    
    This reverts commit 1dcb072cda091180a5b8b03c030bcbe83a54f8e2.

diff --git a/SPIRV/CMakeLists.txt b/SPIRV/CMakeLists.txt
index 61c7a266..286ab73c 100644
--- a/SPIRV/CMakeLists.txt
+++ b/SPIRV/CMakeLists.txt
@@ -70,13 +70,6 @@ set(SPVREMAP_HEADERS
     SPVRemapper.h
     doc.h)
 
-set(PUBLIC_HEADERS
-    GlslangToSpv.h
-    disassemble.h
-    Logger.h
-    spirv.hpp
-    SPVRemapper.h)
-
 add_library(SPIRV ${LIB_TYPE} ${SOURCES} ${HEADERS})
 set_target_properties(SPIRV PROPERTIES
     FOLDER glslang
@@ -150,5 +143,5 @@ if(PROJECT_IS_TOP_LEVEL)
     ")
     install(FILES "${CMAKE_CURRENT_BINARY_DIR}/SPIRVTargets.cmake" DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake)
 
-    install(FILES ${PUBLIC_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/glslang/SPIRV/)
+    install(FILES ${HEADERS} ${SPVREMAP_HEADERS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/glslang/SPIRV/)
 endif()
diff --git a/glslang/CMakeLists.txt b/glslang/CMakeLists.txt
index 37eecaad..10de92a7 100644
--- a/glslang/CMakeLists.txt
+++ b/glslang/CMakeLists.txt
@@ -168,7 +168,7 @@ set(GLSLANG_HEADERS
     Include/SpirvIntrinsics.h
     Include/Types.h)
 
-add_library(glslang ${LIB_TYPE} ${GLSLANG_SOURCES} ${GLSLANG_HEADERS})
+add_library(glslang ${LIB_TYPE} ${GLSLANG_SOURCES} ${GLSLANG_HEADERS} ${BISON_GLSLParser_OUTPUT_SOURCE})
 add_library(glslang::glslang ALIAS glslang)
 set_target_properties(glslang PROPERTIES
     FOLDER glslang
@@ -252,16 +252,12 @@ if(PROJECT_IS_TOP_LEVEL)
         install(FILES "${CMAKE_CURRENT_BINARY_DIR}/glslangTargets.cmake" DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake)
     endif()
 
-    set(PUBLIC_HEADERS
-        Public/ResourceLimits.h
-        Public/ShaderLang.h
-        Public/resource_limits_c.h
-        Include/glslang_c_interface.h
-        Include/glslang_c_shader_types.h
-        Include/ResourceLimits.h
-        MachineIndependent/Versions.h)
+    set(ALL_HEADERS
+        ${GLSLANG_HEADERS}
+        ${MACHINEINDEPENDENT_HEADERS}
+        ${RESOURCELIMITS_HEADERS})
 
-    foreach(file ${PUBLIC_HEADERS})
+    foreach(file ${ALL_HEADERS})
         get_filename_component(dir ${file} DIRECTORY)
         install(FILES ${file} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/glslang/${dir})
     endforeach()
